x-pack/plugin/slm/src/main/java/org/elasticsearch/xpack/slm/action/TransportGetSnapshotLifecycleAction.java

```diff
@@ -80,24 +80,18 @@ protected void masterOperation(
                 );
             }
         } else {
-            final Map<String, SnapshotLifecyclePolicyItem.SnapshotInProgress> inProgress;
-            SnapshotsInProgress sip = state.custom(SnapshotsInProgress.TYPE);
-            if (sip == null) {
-                inProgress = Collections.emptyMap();
-            } else {
-                inProgress = new HashMap<>();
-                for (List<SnapshotsInProgress.Entry> entriesForRepo : sip.entriesByRepo()) {
-                    for (SnapshotsInProgress.Entry entry : entriesForRepo) {
-                        Map<String, Object> meta = entry.userMetadata();
-                        if (meta == null
-                            || meta.get(SnapshotsService.POLICY_ID_METADATA_FIELD) == null
-                            || (meta.get(SnapshotsService.POLICY_ID_METADATA_FIELD) instanceof String == false)) {
-                            continue;
-                        }
-
-                        String policyId = (String) meta.get(SnapshotsService.POLICY_ID_METADATA_FIELD);
-                        inProgress.put(policyId, SnapshotLifecyclePolicyItem.SnapshotInProgress.fromEntry(entry));
+            final Map<String, SnapshotLifecyclePolicyItem.SnapshotInProgress> inProgress = new HashMap<>();
```

Comment:
It seems we lose the case where this is `Collections.emptyMap();` for empty repos. But I'm not sure if that's a common case and we should be concerned with creating a lot of unnecessary empty new hashmaps? I wonder if you could make it an emptyMap for starters, and then initialize a new hashmap at the first entry you see below.

Reply:
This was a weird case where SnapshotsInProgress is `null` that is now replaced by the empty default instance. This case would only happen on clusters that have never ever seen a snapshot run, so the optimisation was pointless :)